### PR TITLE
[StyleCop] Set single, unified ruleset file for all project in the solution

### LINKED ISSUE #1422: Move to a single, unified, .ruleset file
Right now, we have a .ruleset file in each of the libraries in the SDK. 

This should be moved to a single, unified, ruleset file that's configured through the dir.build.props file. 

Note: This work should be done *BEFORE* we move to the "Warnings are Errors" model. This should apply to all library projects, and (maybe) the test projects.

### PR DESCRIPTION
**Pull request by JuanAr**
Fixes #1422 and #1459

Replaces #1429

### Description
In this pull request, we unified all the rules from the projects into a single ruleset file and we placed it at the solution level, we use the Directory.Build.props to set this ruleset file to all the projects in the solution 

### Changes made
Add BotBuilder-DotNet.ruleset file.
Remove the duplicated ruleset files from projects.

### CODE REVIEW

**Comment by cleemullins** [comment]
@JuanAr Proving the value of CI builds, the build is failing. 

Errors are:

> ##[error]tests\integration\Microsoft.Bot.Builder.Integration.AspNet.WebApi.Tests\Microsoft.Bot.Builder.Integration.AspNet.WebApi.Tests.csproj(168,5): Error : This project references NuGet package(s) that are missing on this computer. Use NuGet Package Restore to download them.  For more information, see http://go.microsoft.com/fwlink/?LinkID=322105. The missing file is ..\..\..\packages\xunit.core.2.3.1\build\xunit.core.props.


> ##[error]tests\Microsoft.Bot.Builder.TestBot.WebApi\Microsoft.Bot.Builder.TestBot.WebApi.csproj(185,5): Error : This project references NuGet package(s) that are missing on this computer. Use NuGet Package Restore to download them.  For more information, see http://go.microsoft.com/fwlink/?LinkID=322105. The missing file is ..\..\packages\Microsoft.Net.Compilers.2.6.1\build\Microsoft.Net.Compilers.props.

**Comment by cleemullins** [comment]
Note: It *seems* the master branch is building fine. Per you email, I'm double-checking that now and have a build running. The build that was failing was an experimental build that's about pushing to Azure Artifacts.

File: Directory.Build.props [hunk 1]
```diff
@@ -3,6 +3,14 @@
     <PackageOutputPath>$(MSBuildThisFileDirectory)outputpackages\</PackageOutputPath>
   </PropertyGroup>
 
+  <PropertyGroup>
+    <CodeAnalysisRuleSet>$(SolutionDir)BotBuilder-DotNet.ruleset</CodeAnalysisRuleSet>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="StyleCop.Analyzers" Version="1.0.2" PrivateAssets="all" />
```
Comment (cleemullins) [inline Directory.Build.props]: I don't understand why this is both here, and in each of the .csproj files. Could you elaborate? 

It seems like only 1 location would be needed.

Comment (GasparAcevedoZainSouthworks) [reply Directory.Build.props]: Hi @cleemullins, we update the PR for removing the double reference. Now, the _StyleCop.Analyzers_ NuGet is only referenced in the _Directory.Build.props_ file.

**Pull request merged**